ranger2/aws/ranger_asg/ranger_policy.py

```diff
@@ -0,0 +1,161 @@
+import requests
+import json, sys
+
+#This script takes 4 arguments.
+#Loadbalancer Endpoint.
+#Defloc Location.
+#Service Name to be Created.
+#Loadbalancer Port
+#Usage: python ranger_policy.py LB-IP pritish-qubole hive_policy 80
+
+host = sys.argv[1]
+defloc_loc =  sys.argv[2] 
+service_name = sys.argv[3]
+port = sys.argv[4]
+
+#host = "ps-cvs-ranger-asg-alb-1733501465.us-east-1.elb.amazonaws.com"
+#defloc_loc =  "pritish-qubole" 
+#service_name = "hivedev"
+#port = 80
+
+if port != 80:
+    ranger_url= 'http://' + host + ':' + str(port) + '/service'
+else:
+    ranger_url= 'http://' + host + '/service'
+
+user_url = ranger_url + '/xusers/'
+policy_url = ranger_url + '/public'
+
+groups_url = ranger_url + '/xusers/secure/groups'
+qbl_usr = 'qbol_user'
+
+hdrs={'content-type':'application/json','accept':'application/json'}
+per_page_records = '1000000'
+
+def get_req_data(url, json_data):
+    if json_data is None:
+        json_data = {"page":"1", "per_page": per_page_records}
+    response = requests.get(url, json=json_data, auth=('admin','admin'), headers=hdrs)
+    return(response)
+
+def post_req_data(url, json_data):
+    response = requests.post(url,json=json_data, auth=('admin','admin'), headers=hdrs)
+    return(response)
+
+def put_req_data(url, json_data):
+    response = requests.put(url,json=json_data, auth=('admin','admin'),headers=hdrs)
+    return(response)
+
+def delete_req_data(url):
+    response = requests.delete(url, auth=('admin','admin'),headers=hdrs)
+    return(response)
+
+# Create Service
+def create_service():
+    service_policy_url = policy_url + '/v2/api/service'
+    config_data = {"password":"dummy","jdbc.driverClassName":"org.apache.hive.jdbc.HiveDriver",
+               "jdbc.url":"none","username":"dummyuser"}
+
+    json_data = {"name": service_name, "description": "Hive Qds Policy","type": "hive",
+            "configs": config_data, "isEnabled": True}
+#    print(service_policy_url)
+#    print(json_data)
+    res = post_req_data(service_policy_url, json_data)
+    res_data = json.loads(res.text)
+    if(res.status_code == 200):
+        print('Created service successfully: ' + service_name )
+    else:
+        print('Failed creating Ranger Repository: ' + str(res_data))
+        sys.exit(1)
+
+# Create qbol_user
+def create_qbl_usr(qbl_user):
+    qbol_user_json = {"name": qbl_user,"firstName": qbl_user,"password":"Ranger123","userRoleList":["ROLE_ADMIN_AUDITOR"]} 
```

Comment:
we should probably convert this file to a tpl file and make these configurable. hardcoded passwords might not sit well with folks.

Reply:
Fixed. Added config parameter